lib/chat_widget.dart

```diff
@@ -19,6 +22,35 @@ class SingleChat extends StatefulWidget {
 
 class _SingleChatState extends State<SingleChat> {
   bool actionsExpanded = false;
+  bool imageSelected = false;
+  late File image;
+
+  // Handler for selecting images
+  Future getImage() async {
+    FilePickerResult? result = await FilePicker.platform.pickFiles(
+      type: FileType.custom,
+      allowedExtensions: ['jpg', 'png'],
+    );
+
+    if (result != null) {
+      File file = File(result.files.single.path!);
+      setState(() {
+        image = file;
+        imageSelected = !imageSelected;
+      });
+    } else {
+      setState(() {
+        imageSelected = !imageSelected;
+      });
+    }
```

Comment:
then call setState once here.  The other way works but I would find it easier to read and make sure it isn't called twice this way.